### PR TITLE
CANS-258: Child/Youth Date of Birth Updated

### PR DESCRIPTION
**Pull request by DmitriyRudenko**
## Tests
- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters and static analyzers for the project (SonarQube, eslint, rubocop, etc).
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/cans/domain/dto/PersonDto.java [hunk 1]
```diff
@@ -27,8 +28,11 @@
 @EqualsAndHashCode(callSuper = true)
 @ToString(callSuper = true)
 @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ValidPerson
 public class PersonDto extends Dto {
-  @NotNull private PersonRole personRole;
+
+  @NotNull
```
Comment (edgibbs) [inline src/main/java/gov/ca/cwds/cans/domain/dto/PersonDto.java]: OK, this is a pure Google Style Guide fix, could we perhaps run a fix across the whole codebase as a PR so these don't keep cropping up?

File: src/main/java/gov/ca/cwds/cans/validation/ValidPersonValidator.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+public class ValidPersonValidator implements ConstraintValidator<ValidPerson, PersonDto> {
+
+
+  @Override
+  public void initialize(ValidPerson constraintAnnotation) {
+
+  }
+
+  @Override
+  public boolean isValid(PersonDto value, ConstraintValidatorContext context) {
```
Comment (edgibbs) [inline src/main/java/gov/ca/cwds/cans/validation/ValidPersonValidator.java]: Shouldn't this have a unit level test?
Cases that need coverage:

- null
- past date
- future date and assertion that context has a violation that states "Date of birth must not be a future date"

File: src/main/java/gov/ca/cwds/cans/validation/ValidPerson.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+/**
+ * @author dmitry.rudenko
+ */
```
Comment (edgibbs) [inline src/main/java/gov/ca/cwds/cans/validation/ValidPerson.java]: I know it's probably part of the style guide, but with everything stored in git and projects shared with multiple teams would it be easier to leave off the @author annotations?

File: src/main/java/gov/ca/cwds/cans/validation/ValidPersonValidator.java (hunk 2 above)
Comment (DmitriyRudenko) [reply src/main/java/gov/ca/cwds/cans/validation/ValidPersonValidator.java]: * null validation: ValidPersonValidator.java:110
* error message validation: ValidPersonValidator.java:179
* past dob : ValidPersonValidator.java:251

Comment (DmitriyRudenko) [reply src/main/java/gov/ca/cwds/cans/validation/ValidPersonValidator.java]: @edgibbs  plze review

**Comment by edgibbs** [comment]
I'm pretty sure you're taking about PersonResourceTest for the references, so I'm going to go ahead and approve.

**Pull request merged**